### PR TITLE
Fix linting with Xcode 8

### PR DESCRIPTION
**Pull request by neonichu**
- Uses the new generic `:oldest` device feature in `fourflusher` so that we do not have to hardcode names anymore (great suggestion @orta 🍻)
- Uses an updated version of `fourflusher` which can parse two-digit OS
  versions (hello iOS 10.0)

### CODE REVIEW

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -722,7 +722,7 @@ def xcodebuild
       case consumer.platform_name
       when :ios
         command += %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator)
-        command += Fourflusher::SimControl.new.destination('iPhone 4s', deployment_target)
+        command += Fourflusher::SimControl.new.destination('iPhone 5', deployment_target)
```
Comment (orta) [inline lib/cocoapods/validator.rb]: could four flusher check for sims, and we could define "lowest" in some form at CP level?

Comment (neonichu) [inline lib/cocoapods/validator.rb]: that's a good idea, just need to teach fourflusher about the order of simulators — downside would be needing to update it whenever new simulators are shipped

Comment (orta) [inline lib/cocoapods/validator.rb]: we'd need to do it anyway ATM

Comment (orta) [inline lib/cocoapods/validator.rb]: at least only Four Flusher would need an update then

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -722,13 +722,13 @@ def xcodebuild
...
+        command += Fourflusher::SimControl.new.destination(:oldest, :ios, deployment_target)
       when :watchos
         command += %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
-        command += Fourflusher::SimControl.new.destination('Apple Watch - 38mm', deployment_target)
+        command += Fourflusher::SimControl.new.destination(:oldest, :watchos, deployment_target)
       when :tvos
         command += %w(CODE_SIGN_IDENTITY=- -sdk appletvsimulator)
-        command += Fourflusher::SimControl.new.destination('Apple TV 1080p', deployment_target)
+        command += Fourflusher::SimControl.new.destination(:oldest, :tvos, deployment_target)
       end
```
Comment (orta) [inline lib/cocoapods/validator.rb]: 👍

**Comment by segiddins** [comment]
@neonichu is there still a helpful error if say no compatible tvOS simulator is installed?

**Comment by neonichu** [comment]
@segiddins good point, the error message is not super helpful right now, will update it

**Comment by neonichu** [comment]
There will now be an error message like `Simulator for tvOS 10.0 is not available.` if a certain simulator cannot be found.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            1 Error
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! \nYou can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
    
    
  </tbody>
</table>
  

  

  



Here's an example of your CHANGELOG entry:

```markdown
* WIP: Fix linting with Xcode 8 
  [neonichu](https://github.com/neonichu)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by neonichu** [comment]
I think this should be good to 🚢 now

**Comment by segiddins** [comment]
👍🏻

**Comment by fenglinxiu** [comment]
@orta Your solution works. It helps me much. Thank you. The Total file directory path: /Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.1/lib/cocoapods. And the file does not exist in fourflusher 0.3.0.

**Pull request merged**